libcontainer/cgroups/fs/devices.go

```diff
@@ -78,3 +119,22 @@ func (s *DevicesGroup) Remove(d *cgroupData) error {
 func (s *DevicesGroup) GetStats(path string, stats *cgroups.Stats) error {
 	return nil
 }
+
+func readDevicesExceptDefault(path string) (allowed map[string]Empty, err error) {
+	cgroupData, err := readFile(path, "devices.list")
```

Comment:
the same check won't be possible with cgroup v2.

Would it be possible to detect whether it is updating an existing cgroup in another way and return early if `cgroup.Resources.Devices` is not defined?

Reply:
>

> the same check won't be possible with cgroup v2.
> 
> Would it be possible to detect whether it is updating an existing cgroup in another way and return early if `cgroup.Resources.Devices` is not defined?

Cgroup v2 will use `devices_v2.go` to set configuration, so `readDevicesExceptDefault` won't affect cgroup v2. I'll check if cgroup v2 has interface to query allowed devices.

For current design, I think the answer is no. Because the `update` function only set `cpu, disk or memory`, but cgroup manager only support `Set` interface. This will set and replace all subsystem result.

Reply:
@giuseppe I checked cgroup v2 and found that cgroup v2 will not raise `Operation not permitted when invoking `update` function. So I think this patch is only used for cgroup v1